examples-gallery/plot_ball_rolling_on_spinning_disc.py

```diff
@@ -0,0 +1,620 @@
+        np.sum(Tz3**2)) + free[-1] * weight
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[laenge * num_nodes: (laenge + 1) * num_nodes] = (2.0 * free[-1]
+         * free[laenge * num_nodes: (laenge + 1) * num_nodes])
+    grad[(laenge + 1) * num_nodes: (laenge + 2) * num_nodes] = (2.0 *
+        free[-1] * free[(laenge + 1) * num_nodes: (laenge + 2) * num_nodes])
+    grad[(laenge + 2) * num_nodes: (laenge + 3) * num_nodes] = (2.0 * free[-1]
+       * free[(laenge + 2) * num_nodes: (laenge + 3) * num_nodes])
+    grad[-1] = weight
+    return grad
+
+t0, tf = 0.0, duration
+
+# %%
+# The holonomic constraints must be fullfilled by the state constraints.
+x_start = 7.0
+q2_start = x_start / par_map[r]
+y_start = 7.0
+q1_start = -y_start / par_map[r]
+
+initial_state_constraints = {
+    q1: q1_start,
+    q2: q2_start,
+    q3: 0.0,
+    u1: 0.0,
+    u2: 0.0,
+    u3: 0.0,
+    x: x_start,
+    y: y_start,
+    ux: 0.0,
+    uy: 0.0
+    }
+final_state_constraints = {
+    x: 0.0,
+    y: 0.0,
+    ux: 0.0,
+    uy: 0.0,
+    }
+
+instance_constraints = (
+) + tuple(
+    xi.subs({t: t0}) - xi_val for xi, xi_val in
+    initial_state_constraints.items()
+) + tuple(
+    xi.subs({t: tf}) - xi_val for xi, xi_val in
+    final_state_constraints.items()
+)
+
+# %%
+# Forcing h > 0. helps to avoid negative h solutions.
+
+grenze = 10.0
+bounds = {t1: (-grenze, grenze), t2: (-grenze, grenze),
+    t3: (-grenze, grenze), h: (0.0001, 1.0)}
+
+# %%
+# Create an optimization problem.
+prob = Problem(
+    obj,
+    obj_grad,
+    eom,
+    state_symbols,
+    num_nodes,
+    interval_value,
+    known_parameter_map=par_map,
+    known_trajectory_map = {T(t): interval_fix},
+    instance_constraints=instance_constraints,
+    time_symbol=t,
+    bounds=bounds,
+    )
+
+# %%
+# The initial guess should meet the configuration constrains. It will be
+# plotted below.
+i1b = np.zeros(num_nodes)
+i2 = np.linspace(initial_state_constraints[x],
+    final_state_constraints[x], num_nodes)
+i1a = i2 / par_map[r]
+i3 = np.linspace(initial_state_constraints[y],
+    final_state_constraints[y], num_nodes)
+i1 = -i3 / par_map[r]
+i4 = np.zeros(8*num_nodes)
+initial_guess = np.hstack((i1,i1a, i1b, i2, i3, i4, 0.01))
+fig1, ax1 = plt.subplots(14, 1, figsize=(7.25, 1.25*14), sharex=True)
+prob.plot_trajectories(initial_guess, ax1)
+
+# %%
+# This way the maximum number ofinterations may be changed.
+# Default is 3000.
+prob.add_option('max_iter', 2000)
+
+# %%
+# Find the optimal solution.
+
+solution, info = prob.solve(initial_guess)
+print('message from optimizer:', info['status_msg'])
+print('Iterations needed',len(prob.obj_value))
+print(f'Optimal h = {solution[-1]:.3e}')
+
+## %%
+# PLot the objective value.
+prob.plot_objective_value()
+
+# %%
+# Plot the accuracy of the results.
+prob.plot_constraint_violations(solution)
+
+# %%
+# Plot the results.
+fig1, ax1 = plt.subplots(14, 1, figsize=(7.25, 3.25*14), sharex=True)
```

Comment:
Or maybe it was this one that was too tall.

Use `subplots(..., layout='constrained')`

Reply:
> Or maybe it was this one that was too tall.
> 
> Use `subplots(..., layout='constrained')`

Great keyword, which I did not know. Now the headings may be read even if the plots are small.